File: src/main/java/ru/ensemplix/command/CommandSender.java
package ru.ensemplix.command;

/**
 * Пользователь который отправил команду.
 */
public interface CommandSender {

    /**
     * Отправить сообщение пользователю.
     *
     * @param message Сообщение, которое отправляем пользователю.
     * @param args Аргументы, которые будут подставлены в строку.
     */
    void sendMessage(String message, Object... args);

    /**
     * Может ли пользователь выполнять указанную команду.
     *
     * @param command Команду, которую пытается выполнить пользователь.
     * @param action Действие, которе пытается выполнить пользователь.
     * @return {@code true}, если может выполнять команду.
     */
    boolean canUseCommand(String command, String action);

}

File: src/main/kotlin/ru/ensemplix/command/CommandSender.kt
package ru.ensemplix.command

/**
 * Пользователь который отправил команду.
 */
interface CommandSender {

    /**
     * Отправить сообщение пользователю.
     *
     * @param message Сообщение, которое отправляем пользователю.
     * @param args Аргументы, которые будут подставлены в строку.
     */
    fun sendMessage(message: String, vararg args: Any)

}
